### ISSUE TITLE
i9100 RIL for modems other than LS8

### ISSUE THREAD

**Issue by trincaht** [TheMuppets/proprietary_vendor_samsung#584]
The libsec-ril.so published here is for modem LS8. Is there any way to provide RILs for other modems such as MS1, mostly used in Latin America? THe LS8 modem have a lot of disconnections, poor signal most of the times and 3G 850 is almost never connected. Changing the modem for MS1, does not fix the problem, but then it may be related to a RIL mosmatch. I live in Costa Rica, however recently I took my phone to Argentina (another country with the MS1 modem) and found the same disconnection and poor signal problems.

**Comment by E3V3A** [comment]
AFAICT the "ril" binaries should be independent of the modem. It's a myth that they need to match. The only matching that need to be done is that over completely different networks like CDMA vs LTE, and support of these when modem and HW allows for such support, but then we're no longer speaking of the GT-I9100, but of variations thereof. For example, I'm using an age old modem KI4 on Lollipop and it just works. More importantly, is that the modem (as often extracted from OEM adjusted firmware) is not compatible between operators and countries. So if you're going to supply a modem, make sure it's an *open market* type that supports the all networks in those countries you're planning to use it in. The latest open market modem for Nordic countries (northern Europe) is XX**MS4**. If they're not aimed for the open market, you're likely to end up with a modem with SIM operator locks, bandwidth restrictions, network band limitations (e.g. GSM not 850 but 900 etc) and several other funny behaviors.

**Comment by trincaht** [comment]
Thank-you for your response. You are right in some aspects, for instance, but do not agree in others. It is well known modems are closely related to geographic regions and Radio Frequency Bands, but the RIL layer establishes what bands would be enabled in what regions. According to this site www.worldtimezone.com/gsm.html, MS4 modem used in Nordic Countries, namely Denmark, Finland, Iceland, Norway, Sweden, all use 900, 1800, and 3G 2100 MHz, some others such as Finland use 3G 900/2100, therefore I doubt MS4 modem would be able to connect to 3g 850 MHZ, and GSM 850 1900 as is used in Latin American Countries. However Brazil uses MS1 modem, and looking at the reference table, should cover all 900/1800/850/1900 and 3G 850/2100, 3G 1900/2100. LTE bands are not mentioned as the original GT-I9100 does not cover 4G.
Looking at the functions implemented in the Samsung-RIL project, hosted at redmine.replicant.us/projects/replicant/wiki/Samsung-RIL
there are several functions implemented in the open source Samsung RIL, for instance

RIL_REQUEST_QUERY_NETWORK_SELECTION_MODE 	Get network selection mode 	Working
RIL_REQUEST_SET_NETWORK_SELECTION_AUTOMATIC 	Set automatic network selection mode 	Working
RIL_REQUEST_SET_NETWORK_SELECTION_MANUAL 	Set manual network selection mode 	Working
RIL_REQUEST_QUERY_AVAILABLE_NETWORKS 	Query available networks 	Working 

Now here comes the $1,000,000.00 question: Could the above functions be configured via a text config file such as network_band.conf, and therefore allow to make completely functional the bands already enabled in the modem binary? Or shall we do it at libsamsung-ipc level ?

**Comment by E3V3A** [comment]
> The LS8 modem have a lot of disconnections, poor signal most of the times and 3G 850 is almost never connected.

I don't think we disagree, but there are many other settings you need to be aware of, for example, in what order the networks are acquired and prioritized. If what you say above is true, then you're phone is probably set to prioritize GSM acquisition first and UMTS (3G) second. So if you're in a region of poor 3G signal coverage, your phone will attempt to connect to any GSM it can find. Or worse, you're phone is set to use 3G only, and network is trying to maintain your connections by handover to 2G when 3G signal weakens. Then your phone drops the call.

**Comment by trincaht** [comment]
Mr E3V3A, my respects; as I can see, you are also E:V:A at xda-deveopers comunity, and looking at your contributions, I am under the impression you know a lot more information than it surfaces on the above comments. Yes, you are right in the matter of what networks are connected and prinside ioritized, as sec-csc.zip found in the original ROM, cache.img file, or the vendor customization files that hold the Samsung propietary libsec-ril.so, it also contains other settigs that enable or disable functions in the modem, like customer.xml, holding settings such as Nbfastdormancy <VALUE> or FastDormancyEnableStatus, some other values may be found in build.prop. Another aspect to consider is the fact that since the introduction of GS2 back in 2011, many carriers have adopted 4G bands, and the prioritization of band access may have also changed. So the above leads me to another question: How could we parse values found in late values taken from customer.xml and build.prop into the opensource implemented RIL layer? BTW, I downloaded your XM626GOLD modemHack Pack and found that Very Interesting.
